lib/drivers/oneTouchUltra2.js

```diff
@@ -0,0 +1,386 @@
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the License for more details.
+ *
+ * You should have received a copy of the License along with this program; if
+ * not, you can obtain one from Tidepool Project at tidepool.org.
+ * == BSD2 LICENSE ==
+ */
+
+/**
+ * IOET's code start here
+ * pvalarezo@ioet.com
+ *
+ * The registers looks like:
+ * P "SAT","03/21/15","16:45:24   ","  081 ","N","03", 00 09B6
+ * P "SAT","03/21/15","16:42:11   ","  105 ","B","02", 00 099F
+ * P "SAT","03/21/15","16:39:46   ","  176 ","N","02", 00 09C1
+ *
+ */
+
+var _ = require('lodash');
+//Asyncronous javscript
+var async = require('async');
+//TidePool lib for ISO date handling
+var sundial = require('sundial');
+//TidePool struct lib
+var struct = require('../struct.js')();
+
+module.exports = function (config) {
+  var cfg = _.clone(config);
+  cfg.deviceData = null;
+  var serialDevice = config.deviceComms;
+
+  //BYTE START DEFINITION HEX
+  var DM = '\x11\x0d';
+  var DMDateTime = DM.concat('\x44\x4d\x46');
+  var DMSoftVersion = DM.concat('\x44\x4d\x3f');
+  var DMSerialNumber= DM.concat('\x44\x4d\x40');
+  var DMUpload = DM.concat('\x44\x4d\x50');
+  var DMGlucoseUnits = DM.concat('\x44\x4d\x53\x55\x3f');
+  var DMTimeFormat = DM.concat('\x44\x4d\x53\x54\x3f');
+  var CR = 0x0D;
+  var LF = 0x0A;
+
+  function buildDateTime(date, time) {
+    var fmt = 'MM/DD/YY HH:mm:ss';
+    var ddate = date + ' ' + time;
+    var parsed = sundial.parseFromFormat(ddate, fmt);
+    var dev = sundial.formatDeviceTime(parsed);
+    var utc = sundial.applyTimezone(dev, cfg.timezone).toISOString();
+    var offset = sundial.getOffsetFromZone(utc, cfg.timezone);
+    return { dev: dev, utc: utc, offset: offset };
+  }
+
+  var buildPacket = function (bytes) {
+    var packet = {
+      id: 'OneTouchUltra2 ', // wait for the serial number later
+      bytes: bytes,
+      valid: false,
+      packet_len: 0,
+      lcb: 0,
+      payload: null,
+      crc: 0
+    };
+
+    var packet_len = bytes.byteLength;
+    packet.packet_len = packet_len;
+    packet.valid = true;
+    return packet;
+  };
+
+  var otu2MessageHandler = function (buffer) {
+    if (buffer.len() < 1) {
+      return false;
+    }
+    // there's enough there to try, anyway
+    var packet = buildPacket(buffer.bytes());
+    if (packet.packet_len !== 0) {
+      // remove the now-processed packet
+      buffer.discard(packet.packet_len);
+    }
+
+    if (packet.valid) {
+      return packet;
+    } else {
+      return null;
+    }
+  };
+
+  var otu2ErrorHandler = function(info) {
+    // we don't care too much for timeout errors
+    if(info.error && info.error!=='timeout'){
+      console.log('DEBUG OTU2ERROR:', info);
+    }
+    if (info.connectionId && info.error) {
+      if (info.error == 'timeout') {
+        return -9; // TODO check this
```

Comment:
Returning a value from the error handler does nothing. The only thing you can do here is log the result. See https://developer.chrome.com/apps/serial#event-onReceiveError